domain/application/service/application_test.go

```diff
@@ -976,33 +968,36 @@ func (s *applicationServiceSuite) TestAddUnits(c *gc.C) {
 
 	appUUID := applicationtesting.GenApplicationUUID(c)
 
-	u := application.AddUnitArg{
+	now := ptr(s.clock.Now())
+	u := []application.AddUnitArg{{
 		UnitName: "ubuntu/666",
 		UnitStatusArg: application.UnitStatusArg{
-			AgentStatus: application.UnitAgentStatusInfo{
-				StatusID: application.UnitAgentStatusAllocating,
-				StatusInfo: application.StatusInfo{
-					Since: s.clock.Now(),
-				},
+			AgentStatus: &application.StatusInfo[application.UnitAgentStatusType]{
+				Status: application.UnitAgentStatusAllocating,
+				Since:  now,
 			},
-			WorkloadStatus: application.UnitWorkloadStatusInfo{
-				StatusID: application.UnitWorkloadStatusWaiting,
-				StatusInfo: application.StatusInfo{
-					Message: "installing agent",
-					Since:   s.clock.Now(),
-				},
+			WorkloadStatus: &application.StatusInfo[application.UnitWorkloadStatusType]{
+				Status:  application.UnitWorkloadStatusWaiting,
+				Message: corestatus.MessageInstallingAgent,
+				Since:   now,
 			},
 		},
-	}
+	}}
 	s.state.EXPECT().GetModelType(gomock.Any()).Return("caas", nil)
 	s.state.EXPECT().GetApplicationIDByName(gomock.Any(), "ubuntu").Return(appUUID, nil)
-	s.state.EXPECT().AddUnits(gomock.Any(), appUUID, u).Return(nil)
+
+	var received []application.AddUnitArg
+	s.state.EXPECT().AddUnits(gomock.Any(), appUUID, gomock.Any()).DoAndReturn(func(_ context.Context, _ coreapplication.ID, args []application.AddUnitArg) error {
+		received = args
+		return nil
+	})
 
 	a := AddUnitArg{
 		UnitName: "ubuntu/666",
 	}
 	err := s.service.AddUnits(context.Background(), "ubuntu", a)
 	c.Assert(err, jc.ErrorIsNil)
+	c.Check(received, jc.DeepEquals, u)
```

Comment:
Same here

Reply:
Because the pointers don't match, mocks equality aren't like gc.DeepEquals, they work on ptr equality.